src/python/Utils/Patterns.py

```diff
@@ -11,3 +11,12 @@ def __call__(cls, *args, **kwargs):
             cls._instances[cls] = \
                     super(Singleton, cls).__call__(*args, **kwargs)
         return cls._instances[cls]
+
+
+def getDomainName(urlStr):
+    """
+    Given a URL string, return the domain name.
```

Comment:
I made a mistake that I am always pointing out in others PR, sorry. Let me write a decent docstring for this helper function.